janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -255,8 +255,12 @@ public void bulkRequest(List<ElasticSearchMutation> requests) throws IOException
         }
 
         final StringBuilder builder = new StringBuilder("/_bulk");
-        if (bulkRefresh != null && !bulkRefresh.toLowerCase().equals("false")) {
-            builder.append("?refresh=" + bulkRefresh);
+        if (bulkRefresh != null && !bulkRefresh.toLowerCase().equals("false") && ingestPipeline != null) {
+            builder.append("?refresh=").append(bulkRefresh).append("&pipeline=").append(ingestPipeline);
+        } else if (bulkRefresh != null && !bulkRefresh.toLowerCase().equals("false")) {
+            builder.append("?refresh=").append(bulkRefresh);
+        } else if (ingestPipeline != null) {
+            builder.append("?pipeline=").append(ingestPipeline);
```

Comment:
This might make it hard to add another parameter. Can you implement with one block per option? Below is one way but maybe there's something cleaner?

```java
final Function<StringBuilder,StringBuilder> appendOp = sb -> sb.append(sb.length() == 0 ? "?" : "&");
final StringBuilder builder = new StringBuilder();
if (ingestPipeline != null) {
    appendOp.apply(builder).append("pipeline=" + ingestPipeline);
}
if (bulkRefresh != null && !bulkRefresh.toLowerCase().equals("false")) {
    appendOp.apply(builder).append("refresh=" + bulkRefresh);
}
builder.insert(0, "/_bulk");
```